src/Operation/Find.php

```diff
@@ -418,6 +419,16 @@ private function createQueryOptions(): array
             $options['modifiers'] = is_object($modifiers) ? document_to_array($modifiers) : $modifiers;
         }
 
+        // Ensure no cursor is left behind when limit == batchSize by increasing batchSize
+        if (isset($options['limit'], $options['batchSize']) && $options['limit'] === $options['batchSize']) {
+            assert(is_integer($options['batchSize']));
```

Comment:
Is this necessary for static analysis? The constructor should already enforce that `batchSize` is an integer if set.

Reply:
Yep, this is for static analysis only.